apps/server/src/modules/learnroom/uc/rooms.uc.spec.ts

```diff
@@ -0,0 +1,135 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { NotFoundException, UnauthorizedException } from '@nestjs/common';
+import { courseFactory, taskFactory, userFactory, setupEntities } from '@shared/testing';
+import { Course, EntityId } from '@shared/domain';
+import { CourseRepo, TaskRepo } from '@shared/repo';
+import { RoomsUc } from './rooms.uc';
+
+describe('rooms usecase', () => {
+	let uc: RoomsUc;
+	let courseRepo: CourseRepo;
+	let taskRepo: TaskRepo;
+
+	beforeAll(async () => {
+		await setupEntities();
+	});
+
+	beforeEach(async () => {
+		const module: TestingModule = await Test.createTestingModule({
+			imports: [],
+			providers: [
+				RoomsUc,
+				{
+					provide: CourseRepo,
+					useValue: {
+						// eslint-disable-next-line @typescript-eslint/no-unused-vars
+						findOne(courseId: EntityId, userId?: EntityId): Promise<Course> {
+							throw new Error('Please write a mock for CourseRepo.findOne');
+						},
+					},
+				},
+				{
+					provide: TaskRepo,
+					useValue: {
+						// eslint-disable-next-line @typescript-eslint/no-unused-vars
+						findBySingleParent(courseId: EntityId) {
+							throw new Error('Please write a mock for TaskRepo.findBySingleParent');
+						},
+					},
+				},
+			],
+		}).compile();
+
+		uc = module.get(RoomsUc);
+		courseRepo = module.get(CourseRepo);
+		taskRepo = module.get(TaskRepo);
+	});
+
+	describe('getBoard', () => {
+		it('should get course for roomId', async () => {
+			const user = userFactory.build();
+			const course = courseFactory.buildWithId({ teachers: [user] });
+			const task = taskFactory.finished(user).buildWithId({ course });
+
+			const spy = jest.spyOn(courseRepo, 'findOne').mockImplementation(() => Promise.resolve(course));
+			jest.spyOn(taskRepo, 'findBySingleParent').mockImplementation(() => Promise.resolve([[task], 1]));
+
+			await uc.getBoard(course.id, user.id);
+			expect(spy).toHaveBeenCalledWith(course.id, user.id);
+		});
+
+		it('should fetch published drafts for students', async () => {
+			const user = userFactory.build();
+			const course = courseFactory.buildWithId({ students: [user] });
+			const task = taskFactory.buildWithId({ course });
+			jest.spyOn(courseRepo, 'findOne').mockImplementation(() => Promise.resolve(course));
+
+			const spy = jest.spyOn(taskRepo, 'findBySingleParent').mockImplementation(() => Promise.resolve([[task], 1]));
+
+			await uc.getBoard(course.id, user.id);
+			expect(spy).toHaveBeenCalledWith(course.id, { draft: false });
+		});
+
+		it('should fetch draft tasks for teachers', async () => {
+			const user = userFactory.build();
+			const course = courseFactory.buildWithId({ teachers: [user] });
+			const task = taskFactory.buildWithId({ course });
```

Comment:
similar to the others (for structure purpose) this line should go above line 79?

Reply:
sure.
Also, it would make sense to refactor all this duplication...